File: JavaResistorGrid/src/Node.java
/**
 * Node is a simple object to store the voltage at each node and whether the 
 * node is fixed, i.e. can the voltage change?
 * A node is either dynamic or fixed. If a node is fixed, 
 * it is either a source or drain node.
 */
public class Node {
	private boolean fixed;
	private double voltage;
	/**
	 * constructor to create a Node with initial voltage 'init_v'
	 * and fixed property 't'
	 */
	public Node(boolean t, double init_v){
		fixed = t; voltage = init_v;
	}
	/**
	 * default constructor creates node with voltage 0 and dynamic type
	 */
	Node(){
		this(false, 0.0);
	}
	/**
	 * changes a node type to fixed and changes the voltage to the 10.0 V max.
	 */
	protected void setSource(){
		fixed = true; voltage = 10.0;
	}
	
	/**
	 * changes a node type to fixed and changes the voltage to the 0.0 V min.
	 */
	protected void setDrain(){
		fixed = true; voltage = 0.0;
	}
	/**
	 * changes a node type to dynamic and resets the voltage to 0.0
	 */
	protected void setDynamic(){
		fixed = false; voltage = 0.0;
	}
	/**
	 * "node type" (boolean - true if fixed, false if dynamic)
	 */
	protected boolean isFixed(){
		return fixed;
	}
	
	/**
	 * changes the node voltage to v if the node is not fixed
	 */
	public void setVoltage(double v) throws IllegalArgumentException{
		if (v > 10.0 || v <0.0){ 
			throw new IllegalArgumentException("Voltage must be in the range 0...10.0");
		}
		else if (fixed){
			throw new IllegalArgumentException("Cannot set fixed node's voltage");
		}
		else
		{voltage = v;}
	}
	
	/**
	 * = "node voltage"
	 */
	public double getVoltage(){
		return voltage;
	}
	
	
}

File: C++/ResistorGrid/ResistorGrid/Node.cpp
//
//  Node.cpp
//  ResistorGrid
//
//  Created by Nitin Shyamkumar on 5/19/14.
//  Copyright (c) 2014 Nitin Shyamkumar. All rights reserved.
//

#include "Node.h"
//basic constructor creates a new node
Node::Node (bool t, float volt){
    fixed = t;
    voltage = volt;
}
Node::Node(void){
    Node(false, 0.0);
}

//returns the voltage of a given node
float Node::getVoltage(void){
    return voltage;
}

void Node::setSource(void){
    fixed = true;
    voltage = 10.0;
}

void Node::setDrain(void){
    fixed = true;
    voltage = 0.0;
}

void Node::setDynamic(void){
    fixed = false;
    voltage = 0.0;
}

bool Node::isFixed(void){
    return fixed;
}

void Node::setVoltage(float setTo) throw (int){
    if (setTo > 10.0 || setTo < 0.0){
        throw (-1);
    }
    voltage = setTo;
}


